packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -75,6 +88,116 @@ ModifierButton.propTypes = {
   }).isRequired,
 };
 
+const isBlockActive = (editor, type, level = undefined) => {
+  const { selection } = editor;
+
+  if (!selection) return false;
+
+  let matchCondition;
+
+  switch (type) {
+    case 'paragraph':
+      matchCondition = (n) => n.type === type;
+      break;
+    case 'heading':
+      matchCondition = (n) => n.type === type && n.level === level;
+      break;
+    default:
+      matchCondition = (n) => n.type === type;
+      break;
+  }
+
+  const match = Array.from(
+    Editor.nodes(editor, {
+      at: Editor.unhangRange(editor, selection),
+      match: (n) => !Editor.isEditor(n) && SlateElement.isElement(n) && matchCondition(n),
+    })
+  );
+
+  return match.length > 0;
+};
+
+const toggleBlock = (editor, format, blockType = 'paragraph', level = undefined) => {
+  Transforms.unwrapNodes(editor, {
+    match: (n) => !Editor.isEditor(n) && SlateElement.isElement(n),
+    split: true,
+  });
+
+  let newProperties = {
+    type: blockType,
+    level: level || undefined,
+  };
+
+  Transforms.setNodes(editor, newProperties);
+};
+
+const BlocksDropdown = () => {
+  const editor = useSlate();
+
+  const [optionSelected, setOptionSelected] = React.useState(Object.keys(blockItems)[0]);
+
+  /**
+   * @param {string} optionKey - key of the heading selected
+   */
+  const selectOption = (optionKey) => {
+    toggleBlock(
+      editor,
+      blockItems[optionKey].name,
+      blockItems[optionKey].value.type,
+      blockItems[optionKey].value.level
+    );
+
+    setOptionSelected(optionKey);
+  };
+
+  return (
+    <Select onChange={selectOption} placeholder="Select" value={optionSelected}>
+      {Object.keys(blockItems).map((key) => (
+        <BlockOption
+          key={blockItems[key].name}
+          value={key}
+          label={blockItems[key].label}
+          icon={blockItems[key].icon}
+          onActive={setOptionSelected}
+        />
+      ))}
+    </Select>
+  );
+};
+
+const BlockOption = ({ value, icon, label, onActive }) => {
+  const { formatMessage } = useIntl();
+  const editor = useSlate();
+
+  const isActive = isBlockActive(
+    editor,
+    blockItems[value].value.type,
+    blockItems[value].value.level
+  );
+
+  if (isActive) {
+    onActive(value);
+  }
+
+  return (
```

Comment:
Can you double check the usage for `Option` i don't think that's how you should be passing an icon and the gap you're introducing will not be consistent with all other Selects iirc.

Reply:
you are right, there is the startIcon prop, I can also remove the BlockOption component definition but I prefer to use it just to avoid defining some declarations inside the map function